File: app/src/main/java/no/nav/tilbakemeldingsmottak/rest/feilogmangler/service/FeilOgManglerService.java
package no.nav.tilbakemeldingsmottak.rest.feilogmangler.service;

import jakarta.inject.Inject;
import no.nav.tilbakemeldingsmottak.consumer.email.aad.AzureEmailService;
import no.nav.tilbakemeldingsmottak.model.MeldFeilOgManglerRequest;
import no.nav.tilbakemeldingsmottak.rest.common.epost.HtmlContent;
import org.slf4j.Logger;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import static org.slf4j.LoggerFactory.getLogger;

@Service
public class FeilOgManglerService {

    private static final Logger log = getLogger(FeilOgManglerService.class);

    private final AzureEmailService emailService;

    @Value("${email_nav_support_address}")
    private String emailToAddress;
    @Value("${email_from_address}")
    private String emailFromAddress;

    @Inject
    public FeilOgManglerService(AzureEmailService emailService) {
        this.emailService = emailService;
    }

    public void meldFeilOgMangler(MeldFeilOgManglerRequest request) {
        emailService.sendSimpleMessage(emailToAddress, "Feil/mangel på nav.no meldt via skjema på nav.no", createContent(request));
        log.info("Melding om feil og mangler videresendt til " + emailToAddress);
    }

    private String createContent(MeldFeilOgManglerRequest request) {
        HtmlContent content = new HtmlContent();

        if (request.getOnskerKontakt()) {
            content.addParagraph("Innsender ønsker å kontaktes på epost", request.getEpost());
        }
        content.addParagraph("Hva slags feil", request.getFeiltype().name());
        content.addParagraph("Melding", request.getMelding());

        return content.getContentString();
    }

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/rest/feilogmangler/service/FeilOgManglerService.kt
package no.nav.tilbakemeldingsmottak.rest.feilogmangler.service

import no.nav.tilbakemeldingsmottak.consumer.email.aad.AzureEmailService
import no.nav.tilbakemeldingsmottak.model.MeldFeilOgManglerRequest
import no.nav.tilbakemeldingsmottak.rest.common.epost.HtmlContent
import org.apache.commons.text.StringEscapeUtils
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class FeilOgManglerService(private val emailService: AzureEmailService) {

    private val log = LoggerFactory.getLogger(javaClass)

    @Value("\${email_nav_support_address}")
    private lateinit var emailToAddress: String

    fun meldFeilOgMangler(request: MeldFeilOgManglerRequest) {
        emailService.sendSimpleMessage(
            emailToAddress,
            "Feil/mangel på nav.no meldt via skjema på nav.no",
            createContent(request)
        )
        log.info("Melding om feil og mangler videresendt til $emailToAddress")
    }

    private fun createContent(request: MeldFeilOgManglerRequest): String {
        val content = HtmlContent()
        if (request.onskerKontakt == true) {
            request.epost?.let { content.addParagraph("Innsender ønsker å kontaktes på epost", it) }
        }
        request.feiltype?.value?.let { content.addParagraph("Hva slags feil", it) }
        request.melding?.let { content.addParagraph("Melding", StringEscapeUtils.escapeHtml4(it)) }
        return content.contentString
    }
}
